test/01-retry.t

```diff
@@ -28,5 +28,8 @@ is "$(retry true)" '' 'successful command does not show any output by default'
 set +e; out=$(retry -s 0 false 2>&1); rc=$?; set -e
 like "$out" 'Retrying up to 3 more.*Retrying up to 1' 'failing command retries'
 is $rc 1 'failing command returns no success'
+set +e; out=$(retry -s 1 -e -r 2 false 2>&1); rc=$?; set -e
+like "$out" 'sleeping 1s.*sleeping 2s' 'sleep ammount doubles'
+is $rc 1 'failing command returns no success'
```

Comment:
so above I was using simply `-s 0` to kinda "mock" the sleeping but I am sure we can find a better approach that overrides the actual sleep implementation. Maybe https://github.com/os-autoinst/scripts/blob/master/test/01-label-known-issues.t#L21 gives an idea how to mock sleep but check what argument we actually calculated without actually sleeping?

Reply:
That is a good idea, I will try to implement that.